lib/ts/MemArena.h

```diff
@@ -89,67 +97,74 @@ class MemArena
       @return a MemSpan of the allocated memory.
    */
   MemSpan alloc(size_t n);
 
-  /** Adjust future block allocation size.
-      This does not cause allocation, but instead makes a note of the size @a n and when a new block
-      is needed, it will be at least @a n bytes. This is most useful for default constructed instances
-      where the initial allocation should be delayed until use.
-      @param n Minimum size of next allocated block.
-      @return @a this
-   */
-  self_type &reserve(size_t n);
+  /** Allocate and initialize a block of memory.
 
-  /** Freeze memory allocation.
+      The template type specifies the type to create and any arguments are forwarded to the constructor. Example:
+      @code
+      struct Thing { ... };
+      Thing* thing = arena.make<Thing>(...constructor args...);
+      @endcode
 
-      Will "freeze" a generation of memory. Any memory previously allocated can still be used. This is an
-      important distinction as freeze does not mean that the memory is immutable, only that subsequent allocations
-      will be in a new generation.
+      Do @b not call @c delete an object created this way - that will attempt to free the memory and break. A
+      destructor may be invoked explicitly but the point of this class is that no object in it needs to be
+      deleted, the memory will all be reclaimed when the Arena is destroyed. In general it is a bad idea
+      to make objects in the Arena that own memory that is not also in the Arena.
+  */
+  template <typename T, typename... Args> T *make(Args &&... args);
 
-      If @a n == 0, the first block of next generation will be large enough to hold all existing allocations.
-      This enables coalescence for locality of reference.
+  /** Freeze reserved memory.
 
-      @param n Number of bytes for new generation.
+      All internal memory blocks are frozen and will not be involved in future allocations. Subsequent
+      allocation will reserve new internal blocks. By default the first reserved block will be large
+      enough to contain all frozen memory. If this is not correct a different target can be
+      specified as @a n.
+
+      @param n Target number of available bytes in the next reserved internal block.
       @return @c *this
    */
   MemArena &freeze(size_t n = 0);
 
-  /** Unfreeze memory allocation, discard previously frozen memory.
-
-      Will "thaw" away any previously frozen generations. Any generation that is not the current generation is considered
-      frozen because there is no way to allocate in any of those memory blocks. thaw() is the only mechanism for deallocating
-      memory in the arena (other than destroying the arena itself). Thawing away previous generations means that all spans
-      of memory allocated in those generations are no longer safe to use.
-
-      @return @c *this
+  /** Unfreeze arena.
+   *
+   * Frozen memory is released.
+   *
+   * @return @c *this
```

Comment:
Returns a reference here, not a pointer, right? Same for `freeze`?

Reply:
> Returns a reference here, not a pointer, right? Same for freeze?

I re-read this just now, and this is documenting the return value, not the type, and so it looks correct. Disregard the earlier questions.